src/hooks/useDialog.js

```diff
@@ -0,0 +1,84 @@
+import {useCallback, useEffect} from 'react'
+
+function useDialog({modalRef, isOpen, onDismiss, initialFocusRef, closeButtonRef, returnFocusRef} = {}) {
+  const onClickOutside = useCallback(
+    (e) => {
+      if (modalRef && modalRef.current && !modalRef.current.contains(e.target)) {
```

Comment:
```suggestion
      if (modalRef.current && !modalRef.current.contains(e.target)) {
```

I *think* `modalRef` will always be defined here - even if the `.current` prop is `null`.